packages/cli/src/commands/init/init.ts

```diff
@@ -58,6 +60,30 @@ interface TemplateOptions {
   installCocoaPods?: string | boolean;
 }
 
+const YARN_VERSION = '3.6.4';
+
+const bumpYarnVersion = async (silent: boolean, root: string) => {
+  try {
+    let yarnVersion = semver.parse(getYarnVersionIfAvailable());
+    if (yarnVersion && semver.major(yarnVersion) === 1) {
```

Comment:
Are we checking global yarn version before deciding if we call `yarn set version`? If yes I'd discourage it because the global version is specific to environment. If the project is run on CI or another dev with a different environmnent then it'll result in different behaviour.

e.g. someone has yarn v3 installed globally, creates the project and pushes to GH. then I pull it and run it, but since I have yarn v1 installed globally, it tries to use Yarn v1 and breaks.

To ensure consistent behaviour, we should call this unconditionally.

Reply:
If we're calling it unconditionally, we should first make sure we are not inside a monorepo. We absolutely cannot break existing setups.

Reply:
> If we're calling it unconditionally, we should first make sure we are not inside a monorepo

Sure, tho I don't think the current condition checks anything regarding monorepos so that's a separate thing. With the current check, I could be using a monorepo with Yarn v1 and this will break as well.

Reply:
> it tries to use Yarn v1 and breaks.

Not exactly, because the pulled project has already `.yarnrc.yml` with specified Yarn version so it will use that this one.

> we should first make sure we are not inside a monorepo.

https://github.com/react-native-community/cli/pull/2134/commits/63bec9da3d51e28dc6c3be002a4b70e042edc91a I added logic for checking if we're under git work tree, if yes we're skipping bumping Yarn version.

Reply:
> Not exactly, because the pulled project has already .yarnrc.yml with specified Yarn version so it will use that this one.

But how will the Yarn version be there if it's added conditionally? If I have Yarn 3 installed globally then aren't you skipping `yarn set version`? It's only called if the global Yarn version is 1.

Reply:
Ah okay, I didn't get sense of initial comment. Fixed in https://github.com/react-native-community/cli/pull/2134/commits/bc8aaaa134769e2cc970b016bb567e3725122c7b.